org.eclipse.gef/src/org/eclipse/gef/requests/SimpleFactory.java

```diff
@@ -21,14 +21,14 @@
  */
 public class SimpleFactory implements CreationFactory {
```

Comment:
should we then turn the whole class into a generic? This should not change any client code and future code could better use it.

Reply:
Good point... and done. I also thought about making the whole CreationFactory generic, but I think that'd require quite a lot of changes to resolve all warnings it would introduce...

Reply:
Maybe that is then something for another PR. I general I like these kinds of warnings as they point us to where we can now improve GEF Classic.